pkg/virt-controller/services/template_test.go

```diff
@@ -1030,6 +1030,38 @@ var _ = Describe("Template", func() {
 			Expect(err).To(HaveOccurred())
 		})
 	})
+	Describe("ServiceAccountName", func() {
+
+		It("Should add service account if present", func() {
+			serviceAccountName := "testAccount"
+			vmi := v1.VirtualMachineInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testvmi", Namespace: "default", UID: "1234",
+				},
+				Spec: v1.VirtualMachineInstanceSpec{ServiceAccountName: serviceAccountName, Domain: v1.DomainSpec{}},
+			}
+
+			pod, err := svc.RenderLaunchManifest(&vmi)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(pod.Spec.ServiceAccountName).To(Equal(serviceAccountName), "ServiceAccount matches")
+			Expect(*pod.Spec.AutomountServiceAccountToken).To(BeTrue(), "Token automount is enabled")
```

Comment:
The service account credentials are also mapped inside the VM, correct? Is that something that's realistic to check for here? At the very least, we could inspect the dom XML?

Reply:
The dom XML is tested in `converter_test.go`. Besides that, I have a func test on my todo list, which will test for the token inside the running VM. Does that answer your question?